### ISSUE TITLE
v1 clients aren't trapping a shutdown failure correctly post multi-container

### ISSUE THREAD

**Issue by jhowardmsft** [Microsoft/opengcs#234]
@jterry75 As mentioned offline. I can't see the obvious change, but docker using v1 client is broken for a simple build using initrd.img from master

eg 

```

PS E:\docker\build\36793> docker build --no-cache --platform=linux -t testcachelinux 1
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM alpine
 ---> 3fd9065eaf02
Step 2/2 : RUN echo 1 > /test.txt
 ---> Running in 4cdc545dee6f
The command '/bin/sh -c echo 1 > /test.txt' returned a non-zero code: 4294967295: failed to shutdown container: container 4cdc545dee6fbcca7a9b4f9d238c6a64edc0260f9e7f855f4ff5f5ecad8023cc encountered an error during Shutdown: failure in a Windows system call: Unspecified error (0x80004005)
PS E:\docker\build\36793>
```

```
DEBU[2018-06-27T15:45:05.976009700-07:00] HCSShim::Process::properties succeeded processid=128, properties={"ProcessId":128,"Exited":true,"ExitCode":0,"LastWaitResult":0}
DEBU[2018-06-27T15:45:05.977011600-07:00] HCSShim::Process::ExitCode succeeded processid=128 exitCode=0
DEBU[2018-06-27T15:45:05.977011600-07:00] HCSShim::Process::Close processid=128
DEBU[2018-06-27T15:45:05.977011600-07:00] HCSShim::Process::Close succeeded processid=128
DEBU[2018-06-27T15:45:05.978000300-07:00] HCSShim::Container::Shutdown id=4cdc545dee6fbcca7a9b4f9d238c6a64edc0260f9e7f855f4ff5f5ecad8023cc
DEBU[2018-06-27T15:45:05.979018500-07:00] Result: {"Error":-2147467259,"ErrorMessage":"Unspecified error","ErrorEvents":[{"Message":"a container with the ID \"4cdc545dee6fbcca7a9b4f9d238c6a64edc0260f9e7f855f4ff5f5ecad8023cc\" does not exist","StackTrace":"\ngithub.com/Microsoft/opengcs/service/gcs/core/gcs.(*gcsCore).SignalContainer\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/core/gcs/gcs.go:478\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).signalContainer\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:636\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).shutdownContainer\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:607\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).(github.com/Microsoft/opengcs/service/gcs/bridge.shutdownContainer)-fm\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:238\ngithub.com/Microsoft/opengcs/service/gcs/bridge.HandlerFunc.ServeMsg\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:68\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Mux).ServeMsg\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:139\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).ListenAndServe.func2.1\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:315\nruntime.goexit\n\t/usr/lib/go/src/runtime/asm_amd64.s:2337","Provider":"00000000-0000-0000-0000-000000000000","EventId":0}]}
Inner error:  Unspecified error
DEBU[2018-06-27T15:45:05.980996100-07:00] failed to shutdown container, terminating it  container=4cdc545dee6fbcca7a9b4f9d238c6a64edc0260f9e7f855f4ff5f5ecad8023cc error="container 4cdc545dee6fbcca7a9b4f9d238c6a64edc0260f9e7f855f4ff5f5ecad8023cc encountered an error during Shutdown: failure in a Windows system call: Unspecified error (0x80004005)" module=libcontainerd namespace=moby
```

It something relating to the way GCS is now responding to HCS (or less likely how the HCS "translator" for back-compat is translating the error - almost certainly not this as I would expect to see it using the "old" initrd). 

The old hcsshim is expecting an AlreadyStopped, ElementNotFound (most likely I believe) or ProcNotFound back as valid errors to a shutdown attempt when the container has already exited. Instead, it's getting the "Unspecified error" as can be seen from additional debugging in docker and the fmt.Printf in the additional daemon log above.

**Comment by jhowardmsft** [comment]
You see similar on a simple `docker run -it --rm alpine sh` followed by Ctrl-D, but only in the daemon debug output.  From the CLI perspective, it looks like it worked.


```
DEBU[2018-06-27T16:04:37.715346500-07:00] HCSShim::Process::Wait succeeded processid=128
DEBU[2018-06-27T16:04:37.716350100-07:00] HCSShim::Process::ExitCode processid=128
DEBU[2018-06-27T16:04:37.717347400-07:00] HCSShim::Process::properties processid=128
DEBU[2018-06-27T16:04:37.718332900-07:00] HCSShim::Process::properties succeeded processid=128, properties={"ProcessId":128,"Exited":true,"ExitCode":0,"LastWaitResult":0}
DEBU[2018-06-27T16:04:37.718332900-07:00] HCSShim::Process::ExitCode succeeded processid=128 exitCode=0
DEBU[2018-06-27T16:04:37.718332900-07:00] HCSShim::Process::Close processid=128
DEBU[2018-06-27T16:04:37.719345000-07:00] HCSShim::Process::Close succeeded processid=128
DEBU[2018-06-27T16:04:37.719345000-07:00] HCSShim::Container::Shutdown id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.721346200-07:00] Result: {"Error":-2147467259,"ErrorMessage":"Unspecified error","ErrorEvents":[{"Message":"a container with the ID \"5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a\" does not exist","StackTrace":"\ngithub.com/Microsoft/opengcs/service/gcs/core/gcs.(*gcsCore).SignalContainer\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/core/gcs/gcs.go:478\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).signalContainer\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:636\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).shutdownContainer\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:607\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).(github.com/Microsoft/opengcs/service/gcs/bridge.shutdownContainer)-fm\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:238\ngithub.com/Microsoft/opengcs/service/gcs/bridge.HandlerFunc.ServeMsg\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:68\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Mux).ServeMsg\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:139\ngithub.com/Microsoft/opengcs/service/gcs/bridge.(*Bridge).ListenAndServe.func2.1\n\t/go/src/github.com/Microsoft/opengcs/service/gcs/bridge/bridge.go:315\nruntime.goexit\n\t/usr/lib/go/src/runtime/asm_amd64.s:2337","Provider":"00000000-0000-0000-0000-000000000000","EventId":0}]}
Inner error:  Unspecified error
DEBU[2018-06-27T16:04:37.722321100-07:00] failed to shutdown container, terminating it  container=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a error="container 5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a encountered an error during Shutdown: failure in a Windows system call: Unspecified error (0x80004005)" module=libcontainerd namespace=moby
DEBU[2018-06-27T16:04:37.723352400-07:00] HCSShim::Container::Terminate id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.896332800-07:00] HCSShim::Container::WaitTimeout id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.896332800-07:00] HCSShim::Container::WaitTimeout succeeded id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
WARN[2018-06-27T16:04:37.898331100-07:00] failed to shutdown container                  container=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a error="container 5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a encountered an error during Shutdown: failure in a Windows system call: Unspecified error (0x80004005)" module=libcontainerd namespace=moby process=init
DEBU[2018-06-27T16:04:37.898331100-07:00] HCSShim::Container::Close id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.899341500-07:00] HCSShim::Container::Close succeeded id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
INFO[2018-06-27T16:04:37.899341500-07:00] sending event                                 container=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a event=exit event-info="{5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a init 128 4294967295 2018-06-27 16:04:37.7163501 -0700 PDT m=+1180.941110601 false failed to shutdown container: container 5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a encountered an error during Shutdown: failure in a Windows system call: Unspecified error (0x80004005)}" module=libcontainerd namespace=moby
DEBU[2018-06-27T16:04:37.900329100-07:00] HCSShim::Process::CloseStdin processid=128
DEBU[2018-06-27T16:04:37.900329100-07:00] attach: stderr: end
DEBU[2018-06-27T16:04:37.900329100-07:00] attach: stdin: end
DEBU[2018-06-27T16:04:37.900329100-07:00] attach: stdout: end
DEBU[2018-06-27T16:04:37.901315100-07:00] attach done
DEBU[2018-06-27T16:04:37.901315100-07:00] Revoking external connectivity on endpoint ecstatic_gates (0d5baca0266e6aa59e2ca81b19a46514c6da047140d177fcde2406ba5a539e99)
DEBU[2018-06-27T16:04:37.902332900-07:00] HCSShim::OpenContainer id=5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.914336900-07:00] [DELETE]=>[/endpoints/d01a4a2e-2290-4b18-9ce5-a369e0c49fa4] Request :
DEBU[2018-06-27T16:04:37.927340300-07:00] Releasing addresses for endpoint ecstatic_gates's interface on network nat
DEBU[2018-06-27T16:04:37.927340300-07:00] ReleaseAddress(172.25.128.0/20, 172.25.134.25)
DEBU[2018-06-27T16:04:37.953331200-07:00] lcowdriver: remove: id 5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.954340600-07:00] lcowdriver: remove: id 5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a: layerPath C:\36793\lcow\5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a
DEBU[2018-06-27T16:04:37.956340900-07:00] lcowdriver: remove: id 5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a: layerPath C:\36793\lcow\5dc2f87faa59ab1d67f652ef90eecf238541f58ef341f2fd18c08b5db362297a succeeded
DEBU[2018-06-27T16:04:37.959329300-07:00] Closing buffered stdin pipe
```

**Comment by jhowardmsft** [comment]
Little further digging to confirm - a shutdown when container now running used to return

```
container 90ed01b43ff486cb997fac7e60a3f3371ce17f8c198bf25f5cd7d2147cb26ab9 encountered an error during Shutdown: failure in a Windows system call: The virtual machine or container with the specified identifier is not running. (0xc0370110)
```

**Comment by jhowardmsft** [comment]
Fixed by #237
